docs/user-guide/01-get-started.md

```diff
@@ -0,0 +1,93 @@
+```
+
+If you would like to see the equivalent GLSL of the generated SPIRV code, simply change the `-target` argument to `glsl`:
+```
+.\slangc.exe hello-world.slang -profile glsl_450 -target glsl -o hello-world.glsl -entry computeMain
+```
+
+The resulting `hello-world.glsl` generated by `slangc` is shown below:
+```
+// hello-world.glsl (generated by slangc)
+#version 450
+layout(row_major) uniform;
+layout(row_major) buffer;
+
+#line 2 0
+layout(std430, binding = 0) readonly buffer _S1 {
+    float _data[];
+} buffer0_0;
+
+#line 3
+layout(std430, binding = 1) readonly buffer _S2 {
+    float _data[];
+} buffer1_0;
+
+#line 4
+layout(std430, binding = 2) buffer _S3 {
+    float _data[];
+} result_0;
+
+layout(local_size_x = 1, local_size_y = 1, local_size_z = 1) in;
+void main()
+{
+
+#line 10
+    uint index_0 = gl_GlobalInvocationID.x;
+    float _S4 = ((buffer0_0)._data[(index_0)]);
+
+#line 11
+    float _S5 = ((buffer1_0)._data[(index_0)]);
+
+#line 11
+    float _S6 = _S4 + _S5;
+
+#line 11
+    ((result_0)._data[(index_0)]) = _S6;
+
+#line 8
+    return;
+}
+```
+
+As you can see, things are being translated just as expected to GLSL: the HLSL `StructuredBuffer` and `RWStructuredBuffer` types are mapped to shader storage objects and the `[numthreads]` attribute are translated into proper `layout(...) in` qualifier on the `main` entry-point.
+
+Note that in the generated GLSL code, all shader parameters are qualified with explicit binding layouts. This is because Slang provides a guarantee that all parameters will have fixed bindings regardless of shader optimization. Without generating explicit binding layout qualifiers, the downstream compiler in the driver may change the binding of a parameter depending on whether any preceding parameters are eliminated during optimization passes. In practice this causes a pain in application code, where developers will need to rely on run-time reflection to determine the binding location of a compiled shader kernel. The issue gets harder to manage when the application also needs to deal with shader specializations. Since Slang will always generate explicit binding locations in its output on all targets as if no parameters are eliminated, the user is assured that parameters always gets a deterministic binding location without having to write any manual binding qualifiers in the Slang code themselves. In fact, we strongly encourage users not to qualify their Slang code with explicit binding qualifiers and let the Slang compiler does its work to properly layout parameters. This is best practice to maintain code modularity and avoid potential binding location conflicts between different shader modules.
+
+## The full example
+
+The full Vulkan example that sets up and runs the `hello-world.slang` shader in located in the `/examples/hello-world` directory of the Slang repository. The example code initializes a Vulkan context and runs the compiled SPIRV code.
```

Comment:
It is a little thorny to check in a statement like this that is false at the time it is written.

We probably need some kind of convention in our markdown files for annotating things as callouts/warnings. A fairly simple policy would be to use block quotes where the first line is a header with giving the kind of callout. E.g., the following Markdown:

```
> #### Note ####
> The current `hello-world` implementation does not use raw Vulkan API calls;
> this will be addressed in a future revision so that it matches the documentation.
```

renders as:

> #### Note ####
> The current `hello-world` implementation does not use raw Vulkan API calls;
> this will be addressed in a future revision so that it matches the documentation.

which seems like a reasonable callout for now. If/when we get more serious about how we process our `.md` files into documentation this is the kind of thing that we could detect and customize the formatting for when translating Markdown to HTML.